pkg/operator/operator.go

```diff
@@ -40,16 +52,24 @@ func New(cfg *Config) (*Operator, error) {
 
 	operator.manager, err = manager.New(clientConfig, managerOptions)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to create manager: %v", err)
 	}
 
 	// Setup Scheme for all resources.
 	if err := apis.AddToScheme(operator.manager.GetScheme()); err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to register types: %v", err)
 	}
 
+	// Setup our controllers and add them to the manager.
 	if err := operator.AddControllers(); err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to add controllers: %v", err)
+	}
+
+	// Setup admission webhooks and add them to the manager.
+	if cfg.WebhooksEnabled {
```

Comment:
the manifests are still created so what even when this is disable?

Reply:
You might still run the operator locally and not want the webhooks -- that's just one example. Also, the configuration is set to ignore failure, so even if it exists, you can choose not to enable the webhook.